lib/internal/Magento/Framework/App/Console/MaintenanceModeEnabler.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Class MaintenanceMode * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\App\Console;
+
+use Magento\Framework\App\MaintenanceMode;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class MaintenanceModeEnabler
+ * @package Magento\Framework\App\Console
+ */
+class MaintenanceModeEnabler
+{
+    /**
+     * @var MaintenanceMode
+     */
+    private $maintenanceMode;
+
+    /**
+     * @var bool
+     */
+    private $skipDisableMaintenanceMode;
+
+    /**
+     * @param MaintenanceMode $maintenanceMode
+     */
+    public function __construct(MaintenanceMode $maintenanceMode)
+    {
+        $this->maintenanceMode = $maintenanceMode;
+    }
+
+    /**
+     * Enable maintenance mode
+     *
+     * @param OutputInterface $output
+     * @return void
+     */
+    private function enableMaintenanceMode(OutputInterface $output)
```

Comment:
I made this function private as it changes internal state. So class interfaces with enable/disable methods creates temporal coupling in a code.

Class `MaintenanceModeEnabler` is preserved in a code as it would be wrong to introduce dependency on `OutputInterface` in `MaintenanceMode` (initial recommendation).